### ISSUE TITLE
Updating Lando to Docker Desktop 4.6

### ISSUE THREAD

**Issue by reynoldsalec** [lando/lando#3370]
Docker Desktop 4.6 introduces a new filesharing option called [virtiofs](https://virtio-fs.gitlab.io/) for MacOS users. From early reports, it sounds like many users have enjoyed greatly improved performance from it.

Although users have reported decent experiences updating to Docker Desktop 4.6 and using Lando, there have been some issues. This ticket is an attempt to collect those and solve them.

From what I've seen, we have reported...

- Problems with SSH Key Loading
- Permissions issues: these are known by Docker and should be resolved by them (see https://github.com/docker/roadmap/issues/7#issuecomment-1083310988, but might be worth testing based on [Lando user feedback](https://github.com/lando/lando/issues/763#issuecomment-1087559893)

...I'll update this list as further information becomes clear from users!

**Comment by patrickcate** [comment]
I'd like to see this as well. I'm running into a memory issue building images with M1 MacBook that is apparently fixed in version 4.1+ of Docker Desktop: https://github.com/docker/for-mac/issues/5831

**Comment by notfalsedev** [comment]
+1

**Comment by soniktrooth** [comment]
++ ❤️

**Comment by ayalon** [comment]
Docker 4.6 is buggy with virtiofs. Together with lando only Docker 4.7.1 works.
But mounting the user folder is an issue and SSH_AUTH with Key forwarding does not work properly for now with Lando.
Other than that, the performance gain is amazing.

**Comment by erutan** [comment]
For the very simple use case of running Lando with a Drupal 9 recipe I installed Docker 4.10.1 on MacOS 12.4 and it's extremely fast and so far stable.


If you don't need SSH Key loading and folder mounting - go for it! I'll edit this if I find a need to downgrade to 4.7.

**Comment by reynoldsalec** [comment]
Short update: we have a project to do this that should happen in the next couple months. Apologies for the delay, balancing these needs with Lando 4 development :)

**Comment by andycarlberg** [comment]
Is there a recommended workaround for the SSH key loading? Would love to make the jump but running into the SSH key loading issue.

**Comment by zuernBernhard** [comment]
Docker 4 with virtio fs really feels a lot faster. Here I experience a Problem only in one project with redis. There I can not connect to redis through a socket any more. If a switch of virtio fs and lando rebuild the project it works again with docker 4.12

**Comment by pirog** [comment]
HOLLA! So fundamentally this seems to be a bug(?) in Docker's new experimental `virtiofs` macOS file sharing. The issue is tracked over here https://github.com/docker/for-mac/issues/6243. 

That said there a few ways you can get around this:

1. use the release they posted in this comment https://github.com/docker/for-mac/issues/6243#issuecomment-1257975195, note that it is an EXPERIMENTAL release so while it might resolve this issue it could break other things
2. run lando from source and use this branch https://github.com/lando/cli/tree/load-keys-user-mismatch, note that this only improves SSH key loading and doesnt resolve the deeper root issue eg https://github.com/docker/for-mac/issues/6243

For future reference: In the _still-in-dev_ Lando 4 we are using a different method to sync permissions which should make 6243 a non-issue regardless of whether docker fixes it or not.

**Comment by todeveni** [comment]
@pirog any possibility to land https://github.com/lando/cli/commit/dce7d28a62465caf4fb38c097b6e901c3b934ed7 to upcoming release, since Docker version has been [bumped](https://github.com/lando/cli/commit/e9f36d8cc4bf40a1ce439f4769b2c9951a9fdf03)?

**Comment by pirog** [comment]
@todeveni i dont think we plan to to introduce https://github.com/lando/cli/commit/dce7d28a62465caf4fb38c097b6e901c3b934ed7 since the issue is on the Docker end. It looks like its been resolved as of the latest `4.15.0` Docker Desktop release. However, that release also has this pretty annoying bug in it as well: https://github.com/docker/for-mac/issues/6614

**Comment by reynoldsalec** [comment]
Ok, it looks like the bug in https://github.com/docker/for-mac/issues/6614 has been resolved in Docker Desktop 4.16.1.

We recommend trying out the Lando 3.9.0 release (making sure you actually update to Docker Desktop 4.16.1 to switch to VirtioFS if you're on Mac): https://github.com/lando/lando/releases/tag/v3.9.0

If you're on the [Lando Slack](https://launchpass.com/devwithlando) and have any comments please leave them there, although here works too if that's easier :)

**Comment by xtfer** [comment]
@reynoldsalec Virtiofs works fine, but the new appserver URL tests just time out. I have to quit the startup process or wait for them to time out. Otherwise it works.

**Comment by reynoldsalec** [comment]
@xtfer I'm guessing that's due to the changes made to Lando's [URL scanner](https://docs.lando.dev/core/v3/scanner.html) recently, I'd recommend updating to 3.18+, since we have made those URL tests more liberal in the response codes they "accept" after generating confusion among a bunch of people.
